frontend/src/store/modules/dataOverlay.js

```diff
@@ -0,0 +1,139 @@
+import dataOverlayApi from '@/api/dataOverlay';
+import { DATA_TYPES_COMPONENTS, parseFile } from '@/helpers/dataOverlay';
+import { getSingleExpressionColor } from '@/helpers/expressionSources';
+
+const data = {
+  index: {},
+  currentDataType: null,
+  currentDataSource: null,
+  customDataSource: null,
+  customDataSet: 'None',
+  dataSet: 'None',
+};
+
+const getters = {
+  queryParams: state => ({
+    datatype: state.currentDataType ? state.currentDataType.name : 'None',
+    datasource: state.currentDataSource ? state.currentDataSource.filename : 'None',
+    dataSet: state.dataSet,
+  }),
+  computedLevels: (state) => {
+    const { dataSet, currentDataSource, customDataSource, customDataSet } = state;
+    let t;
+    let l;
+
+    if (customDataSource && customDataSet !== 'None') {
+      t = customDataSet;
+      l = customDataSource.levels;
+    } else if (currentDataSource && dataSet !== 'None') {
+      t = dataSet;
+      l = currentDataSource.levels;
+    } else {
+      return {};
+    }
+
+    const computedLevels = {
+      'n/a': [getSingleExpressionColor(NaN), 'n/a'],
+    };
+
+    Object.keys(l[t]).forEach((id) => {
+      const val = Math.round((l[t][id] + 0.00001) * 100) / 100;
+      computedLevels[id] = [getSingleExpressionColor(val), val];
+    });
+
+    return computedLevels;
+  },
+  componentClassName: state => state.currentDataType && state.currentDataType.className,
+  componentDefaultColor: state => state.currentDataType && state.currentDataType.defaultColor,
+  componentType: state => state.currentDataType && state.currentDataType.componentType,
+};
+
+const actions = {
+  async getIndex({ commit }, model) {
+    const index = await dataOverlayApi.fetchIndex(model);
+
+    commit('setIndex', index);
+  },
+  async setCurrentDataType({ commit, dispatch, state }, { model, type, propagate }) {
+    const currentDataType = {
+      name: type,
+      ...DATA_TYPES_COMPONENTS[type],
+    };
+    commit('setCurrentDataType', currentDataType);
+
+    if (propagate) {
+      const { filename } = state.index[type][0];
+      await dispatch('getDataSource', { model, type, filename, propagate });
+    }
+  },
+  async getDataSource({ commit, dispatch }, { model, type, filename, propagate }) {
+    try {
+      if (propagate) {
+        dispatch('setDataSet', 'None');
+      }
+
+      const file = await dataOverlayApi.fetchFile({
+        model,
+        type,
+        filename,
+      });
+      const dataSource = await parseFile(file);
+      const metadata = data.index[type].find(
+        m => m.filename === filename
+      );
+      commit('setCurrentDataSource', {
+        ...metadata,
+        ...dataSource,
+      });
+    } catch (e) {
+      console.error(e);
```

Comment:
I see a warning message ```warning: Unexpected console statement (no-console) at src/store/modules/dataOverlay.js:89:7:``` with console.log() here.

Reply:
Good catch. I have three ideas:

- adding  `// eslint-disable-line no-console` (lazy solution)

- replace the logging with generating a `<ErrorPanel :message="errorMessage" @hideErrorPanel="errorMessage=''" />` in the component, since it is should be more of a server problem that user action related
- remove the logging and settle with having ` commit('setCurrentDataSource', null);`

Any thoughts?

Reply:
As you mentioned, this is more a server problem, so users are probably not expected to see this error message. We just tell them something like "file uploading failed, please try again". This error message is more for the developers. In this case, we could add a DEBUG mode and only when the DEBUG mode is activated, this error message will be shown in the log. I think it should be done in a separate issue if we are going add this feature. For the moment, before we have a good reason to remove this error catching, I would opt for the solution 1.

Reply:
So far we don't have any other `try catch` in the store, so for simplicity I would continue with this.
Making sure our API responds well is a valid concern, so we should inform the user when that doesn't happen. This is already implemented in other parts of the interface, but not consistently.
My suggestion would be to open up a different issue for this.

Reply:
Sounds good!

Reply:
Issue #691 created for this and commit 4f9a3ea528c69deb9c1c95987bd3c27594caae42 disables the eslint for now